thorin/analyses/deptree.cpp

```diff
@@ -26,9 +26,21 @@ VarSet DepTree::run(Def* nom) {
     auto parent = root_.get();
     for (auto var : result) {
         auto n = nom2node_[var->nom()].get();
+        if (!n) {
+            const_cast<World&>(world_).ELOG("var {} used before nom {} discovered, old var still around?", var,
+                                            var->nom());
+            const_cast<World&>(world_).ELOG("var {} : {}", var, var->type());
+            const_cast<World&>(world_).ELOG("var nom {} : {}", var->nom(), var->nom()->type());
```

Comment:
I think we should remove the `const_cast` and instead declare `World::State::POD::log` as `mutable`.

Reply:
See my latest commit on `master`